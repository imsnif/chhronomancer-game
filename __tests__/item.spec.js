import 'react-native'
import React from 'react'
import Item from '../components/item'
import renderer from 'react-test-renderer'

test('Item (default props) => renders correctly', () => {
  const tree = renderer.create(
    <Item />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Item (with name) => renders correctly', () => {
  const tree = renderer.create(
    <Item name='assistPrevent' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Item (with fill) => renders correctly', () => {
  const tree = renderer.create(
    <Item fill />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Item (with name and fill) => renders correctly', () => {
  const tree = renderer.create(
    <Item name='assistPrevent' fill />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
