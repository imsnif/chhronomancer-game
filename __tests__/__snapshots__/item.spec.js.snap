// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item (default props) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "borderColor": undefined,
      "borderWidth": 1,
      "height": 9,
      "margin": 4,
      "transform": Array [
        Object {
          "rotate": "45deg",
        },
      ],
      "width": 9,
    }
  }
/>
`;

exports[`Item (with color and fill) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "borderColor": "red",
      "borderWidth": 1,
      "height": 9,
      "margin": 4,
      "transform": Array [
        Object {
          "rotate": "45deg",
        },
      ],
      "width": 9,
    }
  }
/>
`;

exports[`Item (with color) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "borderColor": "red",
      "borderWidth": 1,
      "height": 9,
      "margin": 4,
      "transform": Array [
        Object {
          "rotate": "45deg",
        },
      ],
      "width": 9,
    }
  }
/>
`;

exports[`Item (with fill) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderWidth": 1,
      "height": 9,
      "margin": 4,
      "transform": Array [
        Object {
          "rotate": "45deg",
        },
      ],
      "width": 9,
    }
  }
/>
`;
